src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -210,6 +214,60 @@ def shutdown(self):
         self.log.info('Stopping pg_autoscaler')
         self._shutdown.set()
 
+    def check_pool_activity(check_time, pool_stats, pool_id, final_pg_target, old_values):
```

Comment:
I'm still working through how exactly this works, it looks like old_values is basically a dict of pool counters that you first initialize with a default pool id "1" in PgAutoscaler, then you periodically update the counters for every pool when _get_pool_status() is called in _command_autoscale_status() and at regular intervals via _maybe_adjust().  What I'm still a little foggy on is what happens in the check_time true/false cases here.  In both cases we adjust the counters in various ways but it's not clear to me exactly what's going on.  At least an overview of the algorithm and what happens in each conditional would be a helpful comment.

I'm a bit wary of the overhead with the nested loops walking through the entire pool list for every pool and then further iterating through the add/unused counters are.  Have you done any testing to see if this ends up introducing any significant overhead as the number of pools grows? (and at what point it becomes a problem?).  It might not actually matter until you get to a very high number of pools, but my spidey sense is tingling and I suspect there may be some ways to make this faster and less complex.